healthcheck/client_test.go

```diff
@@ -0,0 +1,59 @@
+package healthcheck
+
+import (
+	"errors"
+	"net/http"
+	"testing"
+
+	"github.com/ONSdigital/go-ns/healthcheck/mock_healthcheck"
+	. "github.com/smartystreets/goconvey/convey"
+)
+
+func TestUnitHealthcheckClient(t *testing.T) {
+
+	service := "myService"
+
+	Convey("Healthcheck returns nil when no error", t, func() {
+		mock := &mock_healthcheck.HttpClientMock{
+			GetFunc: func(url string) (*http.Response, error) {
+				return &http.Response{StatusCode:http.StatusOK}, nil
+			},
+		}
+
+		client := healthcheckClient{Client:mock, Url:"http://foo/bar", Service: service}
+
+		s, e := client.Healthcheck()
+		So(s, ShouldEqual, service)
+		So(e, ShouldBeNil)
+	})
+
+	Convey("Healthcheck returns error when status != 200", t, func() {
```

Comment:
This may also be a matter of preference but I find it easier to work with the tests if you put each top level `Convey` in a separate function. When you are debugging tests its much nicer being able to run them isolated.